### PR TITLE
fix and refactor print type result at top level

### PR DESCRIPTION
**Pull request by capfredf**
1. only subtypes of Number are generalized at top level
2. move the code to tc-result

### CODE REVIEW

File: typed-racket-lib/typed-racket/typecheck/toplevel-trampoline.rkt [hunk 1]
```diff
@@ -157,28 +158,37 @@
...
-                           (format "\n[more precisely: ~a]"
-                                   (pretty-format-rep (make-Values (map -result tcs))
-                                                       #:indent 17))]
-                          [else (format " [more precisely: ~a]" (cons 'Values tcs))])
+                          [supertype?
+                           (cond
+                             [indented?
+                              (format "\n[more precisely: ~a]"
+                                      (pretty-format-rep (make-Values (map -result tcs))
+                                                         #:indent 17))]
```
Comment (samth) [inline typed-racket-lib/typed-racket/typecheck/toplevel-trampoline.rkt:189]: Can 17 be calculated?

File: typed-racket-lib/typed-racket/typecheck/toplevel-trampoline.rkt [hunk 2]
```diff
@@ -157,28 +158,37 @@
             ;; before printing them.
             (define tc (cleanup-type t))
             (define tg (generalize tc))
```
Comment (samth) [inline typed-racket-lib/typed-racket/typecheck/toplevel-trampoline.rkt:160]: Should these happen in the other order?

File: typed-racket-test/unit-tests/interactive-tests.rkt [hunk 3]
```diff
@@ -163,6 +163,18 @@
...
       (:print-type (+ 1 1)))
+
+    (test-form #rx"more precisely: Positive-Byte"
+               10)
+
+    (test-form #rx".*(?!more precisely)"
+               (begin (: a-pair (MPairof String Null))
+                      (define a-pair (mcons "123456" '()))
+                      a-pair))
+    (test-form #rx".*\\)\\s*(?!\\[more precisely)"
```
Comment (samth) [inline typed-racket-test/unit-tests/interactive-tests.rkt:174]: Probably this test should specifically rule out mentioning `Mutable-Vectorof`.

File: typed-racket-lib/typed-racket/typecheck/toplevel-trampoline.rkt (hunk 2 above)
Comment (capfredf) [reply typed-racket-lib/typed-racket/typecheck/toplevel-trampoline.rkt]: `cleanup-type` is intended for function types and generalize doesn't deal with function types, so I don't think the order matters in this case.

**Review by samth — APPROVED** [review]

**Pull request merged**